src/server/ApiMessage.cpp

```diff
@@ -153,7 +153,7 @@ HRESULT _CONSOLE_API_MSG::ReleaseMessageBuffers()
 
     if (State.InputBuffer != nullptr)
     {
-        delete[] State.InputBuffer;
+        delete[] static_cast<BYTE**>(State.InputBuffer);

```

Comment:
Is `BYTE**` a correct type here?

Reply:
The only time InputBuffer is assigned is from `State.InputBuffer = pPayload.release();`, and `pPayload` is of type `wistd::unique_ptr<BYTE[]>`

Reply:
`std::unique_ptr<T[]>` means that the pointer to the allocated memory will be deleted with `operator delete[]` by default. And the underlying type is still `T*`, in this case it is `BYTE*` and not `BYTE**`. See short sample program here https://ideone.com/P8kqXK

I'm not sure, but hope that `wistd:: unique_ptr` do the same thing.

Reply:
You're right; I should have checked the return type for `pPayload.release()`.